### PR TITLE
Improve build preview message

### PR DESCRIPTION
**Pull request by Gudahtt**
The build preview message now includes a link to the build preview instructions, and a collapsible list of all published packages.

### CODE REVIEW

File: scripts/generate-preview-build-message.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const rawManifest = await fs.promises.readFile(manifestPath, {
+      encoding: 'utf8',
+    });
+    const { name, version } = JSON.parse(rawManifest);
+
+    packageMap[name] = version;
+  }
+
+  const previewBuildMessage = `
+Preview builds have been published. [See these instructions]() for more information about preview builds.
```
Comment (Gudahtt) [inline scripts/generate-preview-build-message.ts:33]: Link to build instructions to be added here

**Comment by Gudahtt** [comment]
Preview builds have been published. [See these instructions]() for more information about preview builds.

<details>

<summary>Expand for full list of packages and versions:</summary>


```
{
  "@metamask/address-book-controller": "1.0.0-preview.45236158",
  "@metamask/announcement-controller": "1.0.0-preview.45236158",
  "@metamask/approval-controller": "1.0.0-preview.45236158",
  "@metamask/assets-controllers": "1.0.0-preview.45236158",
  "@metamask/base-controller": "1.0.0-preview.45236158",
  "@metamask/composable-controller": "1.0.0-preview.45236158",
  "@metamask/controller-utils": "1.0.0-preview.45236158",
  "@metamask/ens-controller": "1.0.0-preview.45236158",
  "@metamask/gas-fee-controller": "1.0.0-preview.45236158",
  "@metamask/keyring-controller": "1.0.0-preview.45236158",
  "@metamask/message-manager": "1.0.0-preview.45236158",
  "@metamask/network-controller": "1.0.0-preview.45236158",
  "@metamask/notification-controller": "1.0.0-preview.45236158",
  "@metamask/permission-controller": "1.0.0-preview.45236158",
  "@metamask/phishing-controller": "1.0.0-preview.45236158",
  "@metamask/preferences-controller": "1.0.0-preview.45236158",
  "@metamask/rate-limit-controller": "1.0.0-preview.45236158",
  "@metamask/subject-metadata-controller": "1.0.0-preview.45236158",
  "@metamask/transaction-controller": "1.0.0-preview.45236158"
}
```

</details>

**Comment by Gudahtt** [comment]
^ The comment should look something like that

File: scripts/generate-preview-build-message.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Write a preview build message to the path "preview-build-message.txt".
+ */
+async function main() {
+  const packageMap: Record<string, string> = {};
+
+  const packagesDirectory = path.resolve(__dirname, '../packages');
```
[the remark this answers is not in the document]
Comment (Gudahtt) [reply scripts/generate-preview-build-message.ts:12]: Good idea! I have just discovered that `--no-private` isn't supported by this version of Yarn, but it is in the current stable version. It has been updated in this PR: https://github.com/MetaMask/controllers/pull/1012

I can update this to use `yarn workspaces list --no-private` once that has been merged.

File: scripts/generate-preview-build-message.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    packageMap[name] = version;
+  }
+
+  const previewBuildMessage = `
+Preview builds have been published. [See these instructions](https://github.com/MetaMask/controllers/blob/main/docs/contributing.md#using-controllers-in-other-projects-during-developmenttesting) for more information about preview builds.
+
+<details>
+
+<summary>Expand for full list of packages and versions:</summary>
```
Comment (Mrtenz) [inline scripts/generate-preview-build-message.ts:37]: Nit: I think it makes more sense to have a period here. Especially when collapsed, the `:` doesn't make sense.

```suggestion
<summary>Expand for full list of packages and versions.</summary>
```

File: scripts/generate-preview-build-message.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  const packageMap: Record<string, string> = {};
+
+  const { stdout } = await execa('yarn', [
+    'workspaces',
+    'list',
+    '--no-private',
+    '--json',
+  ]);
+  const packages = stdout.split('\n').map((line) => JSON.parse(line));
+  const packageManifestsPaths = packages.map(({ location }) =>
```
Comment (mcmire) [inline scripts/generate-preview-build-message.ts:20]: Nit: `packageManifestsPaths` -> `packageManifestPaths`

**Review by mcmire — DISMISSED** [review]
Looks good!

File: scripts/generate-preview-build-message.ts (hunk 4 above)
Comment (Gudahtt) [reply scripts/generate-preview-build-message.ts]: Fixed here: https://github.com/MetaMask/controllers/pull/984/commits/ff95cc5315ba734fa189fed434c3cffd232cb04f

**Review by mcmire — APPROVED** [review]

**Pull request merged**